File: simple-stack-example-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/util/SchedulerHolder.java
package com.zhuinden.simplestackdemoexamplefragments.util;

import javax.inject.Inject;

import io.reactivex.Scheduler;


/**
 * Created by Owner on 2017. 01. 26..
 */
// NOT SCOPED!
public class SchedulerHolder {
    private Scheduler scheduler;

    @Inject
    public SchedulerHolder() {
    }

    public Scheduler getScheduler() {
        if(scheduler == null) {
            throw new IllegalStateException("Scheduler should not be null!");
        }
        return scheduler;
    }

    public void setScheduler(Scheduler scheduler) {
        this.scheduler = scheduler;
    }
}

File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/util/SchedulerHolder.kt
package com.zhuinden.simplestackdemoexamplefragments.util

import io.reactivex.Scheduler
import javax.inject.Inject


/**
 * Created by Zhuinden on 2018. 08. 20.
 */
// NOT SCOPED!
class SchedulerHolder @Inject constructor() {
    lateinit var scheduler: Scheduler
}
